openquake/hazardlib/source/base.py

```diff
@@ -40,6 +40,8 @@ class BaseSeismicSource(with_metaclass(abc.ABCMeta)):
     __slots__ = ['source_id', 'name', 'tectonic_region_type',
                  'trt_model_id', 'weight', 'seed', 'id']
 
+    MODIFICATIONS = abc.abstractproperty()
```

Comment:
Not sure of the best approach here. I'm following the example in the MFD base class (https://github.com/gem/oq-hazardlib/blob/master/openquake/hazardlib/mfd/base.py#L33). I can change it if necessary.